### PR TITLE
#167190534 Implement  User Create Article

### PR DESCRIPTION
**Pull request by Adekoreday**
#### What does this PR do?
This PR enables a registered to create an article

#### Description of Task proposed in this pull request?
an article model was added 
an article controller was added
article input validations were added
article routes were added

The above were added to provision a persistent endpoint for storing articles
 
#### How should this be manually tested (Quality Assurance)?
-npm install
-npm run start:dev (development server)
-npm run test (test)

#### What are the relevant pivotal tracker stories?
- [167190534](https://www.pivotaltracker.com/story/show/167190534)

#### Any background context you want to add (Operations Impact)?
-set up database sequelize connection
-npm run db:rollback.

#### What I have learned working on this feature:
I learned about sequelize association
I learnt about sequelize slug
I learnt about uploading image to cloudinary

#### Screenshots: [If you made some visual changes to the application please upload screenshots here, or remove this section]
Screenshot 2019-08-14 at 6 49 07 PM
Screenshot 2019-08-15 at 8 53 24 AM

### CODE REVIEW

**Comment by henryade** [comment]
> #### What does this PR do?
> This PR enables a registered to create an article
> 
> #### Description of Task proposed in this pull request?
> an article model was added
> an article controller was added
> article input validations were added
> article routes were added
> 
> The above were added to provision a persistent endpoint for storing articles
> 
> #### How should this be manually tested (Quality Assurance)?
> -npm install
> -npm run start:dev (development server)
> -npm run test (test)
> 
> #### What are the relevant pivotal tracker stories?
> * [167190534](https://www.pivotaltracker.com/story/show/167190534)
> 
> #### Any background context you want to add (Operations Impact)?
> -set up database sequelize connection
> -npm run db:rollback.
> 
> #### What I have learned working on this feature:
> I learned about sequelize association
> I learnt about sequelize slug
> I learnt about uploading image to cloudinary
> 
> #### Screenshots: [If you made some visual changes to the application please upload screenshots here, or remove this section]
> Screenshot 2019-08-14 at 6 49 07 PM
> 
> Screenshot 2019-08-15 at 8 53 24 AM

**Comment by henryade** [comment]
In the screenshot part of your Pull Request message, remove the text in the square bracket because it's just an additional comment.

File: package.json [hunk 1]
```diff
@@ -15,7 +15,7 @@
     "db:seed": "sequelize db:seed:all",
     "start": "node dist/index.js",
     "start:dev": "DEBUG=dev nodemon --exec babel-node server",
-    "test": "npm run db:rollback && nyc mocha --timeout 10000 -r esm --exit"
+    "test": "npm run db:rollback && nyc mocha --timeout 20000 -r esm --exit"
```
Comment (henryade) [inline package.json]: please leave this timeout as 10000

File: server/controllers/Article.js [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import model from '../database/models';
+import { imageUpload, serverResponse } from '../helpers';
+
+const { Article } = model;
+/**
+ * @export
+ * @class Auth
```
Comment (henryade) [inline server/controllers/Article.js:7]: This should be the same with the class name and also the `member of` in the js doc below

**Comment by henryade** [comment]
In comparing your screenshot to the PR specification, I discovered the following fields are missing
`favorited`, `favoritesCount` and authors details with username, bio etc. Can I ask why?

File: server/controllers/Article.js [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+ */
+class Articles {
+  /**
+   * @name createArticle
+   * @async
+   * @static
+   * @memberof Auth
```
Comment (chialuka) [inline server/controllers/Article.js:14]: Please, what is `Auth` that `createArticle` is a member or?

File: server/controllers/Article.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+   * @param {Object} req express request object
+   * @param {Object} res express response object
+   * @returns {JSON} JSON object with details of new user
+   */
+  static async createArticle(req, res) {
+    let image;
```
Comment (chialuka) [inline server/controllers/Article.js:20]: Why aren't you using a `try-catch` in this controller?

**Comment by Adekoreday** [comment]
> In comparing your screenshot to the PR specification, I discovered the following fields are missing
> `favorited`, `favoritesCount` and authors details with username, bio etc. Can I ask why?

**Comment by Adekoreday** [comment]
> In comparing your screenshot to the PR specification, I discovered the following fields are missing
> `favorited`, `favoritesCount` and authors details with username, bio etc. Can I ask why?

the favorited and favoritesCount was changed to dislikesCount and likesCount due to abdul task.
authordetails will be returned when the we get a single  article, which is beyound the scope of this task

**Pull request closed without merging**